### PR TITLE
Support symfony 3.0

### PR DESCRIPTION
**Pull request by pawaclawczyk**
Hi!

I've added some fixes to support for more Symfony versions (2.5, 2.7, 2.8, 3.0). Probably it's not the cleanest way to do that, so I will appreciate any hints and suggestions.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -29,11 +41,13 @@ script:
...
-      vendor/bin/behat --tags=symfony27
+    if [ "$SYMFONY_VERSION" ==  "2.5.*" ]; then
+          vendor/bin/behat --tags=~symfony27
+    else
+          vendor/bin/behat
     fi
 
 notifications:
-  email: matthiasnoback@gmail.com
+  email: p.a.waclawczyk@gmail.com
```
Comment (EmanueleMinotto) [inline .travis.yml:51]: this change should be reverted

**Comment by matthiasnoback** [comment]
Well done! I'll do my best to check this out soon - it looks great at first sight!

**Comment by matthiasnoback** [comment]
Thanks for looking into this! Also, great work in pointing out all the conflicting areas. I think this needs some more thinking though. I somehow sense there should be a simpler solution in some places. Also, there's a difference in making the tests pass and providing a working solution for any existing production application usage of this library. I'll checkout your branch and see if I can figure it out.

**Comment by XanderEmu** [comment]
So... how is it going?

**Comment by matthiasnoback** [comment]
Sorry to all of you! I didn't take the time to look at it. Will try to do so soon.

**Comment by matthiasnoback** [comment]
@pawaclawczyk I finally looked another time at your proposed changes. I think it is too much to carry for this small lib to be compatible with both older and newer versions of Symfony. I propose to prepare a new major release with only Symfony 2.8 and 3.*-compatible code. What do you think? If every agrees, I'll make sure to release that new version (v2.0.0). We don't need to worry as much about BC breaks in that case :)

**Comment by matthiasnoback** [comment]
@pawaclawczyk Really sorry for this to take so long. New job and everything... Well, I picked some of your solutions and as explained above decided to only support Symfony 2.8 and up from now on. I hope everything still works for you. Let me know if anything is wrong.

**Pull request merged**